redaxo/src/addons/structure/plugins/content/boot.php

```diff
@@ -105,3 +105,5 @@
         return rex_url::backendPage($urls[$match[1]][0], ['function' => 'edit', $urls[$match[1]][1] => $match[2]]);
     }
 });
+
+rex_minibar::factory()->addElement(new rex_minibar_structure_content_frontend());
```

Comment:
@staabm Wir sind uns unsicher, was hier das beste wäre.
Aktuell ist die Methode nicht statisch, sie speichert aber in einer statischen Property in der Klasse.
Somit wäre hier eine statische Methode eher angebracht, oder?
Denn factory() soll ja immer ein neues Element liefern, oder?

Und dann stellt sich noch die Frage, ob man das Element-Objekt übergeben sollte, oder nur den Klassennamen.
An anderen solchen Stellen übergeben wir teils nur den Klassennamen, um nicht unnötig das Objekt zu erzeugen, wenn es gar nicht benötigt wird (hier: Minibar inaktiv)
Oder sollte man selbst ein `if (rex_minibar::isActive())` drum rum setzen?

Reply:
Sollten wir statt der factory vllt mit dem singleton arbeiten?

https://github.com/redaxo/redaxo/blob/479fc0fd0be39d2593b7301f4910531fdb9ce3eb/redaxo/src/core/lib/base/singleton_trait.php

Ich würde keine statischen methoden verwenden (auch für setVisibility etc)

Reply:
(Anstatt factory)

Reply:
> Und dann stellt sich noch die Frage, ob man das Element-Objekt übergeben sollte, oder nur den Klassennamen.

Objekte übergeben ist flexibler dür den aufrufer (und ich sehe keinen grund warum wir die klasse übergeben sollten)

Reply:
> Sollten wir statt der factory vllt mit dem singleton arbeiten?

Habe es hier mal umgesetzt 9b9128835b15fbf12f9469eeafad65a75cecd07f